src/logservice/logminer/ob_log_miner_record.cpp

```diff
@@ -594,7 +595,7 @@ int ObLogMinerRecord::build_insert_stmt_(ObStringBuffer &stmt,
             "col_idx", i);
       }
       if (OB_SUCC(ret)) {
-        if (is_lob_type_(col_meta) && nullptr == new_cols[i].buf) {
+        if (is_lob_type_(col_meta) && nullptr == new_cols[i].buf && new_cols[i].m_origin != VALUE_ORIGIN::REDO) {
```

Comment:
why use `!= VALUE_ORIGIN::REDO`, there are other enum value in `VALUE_ORIGIN`

Reply:
if you still use `has_lob_null` as the flag variable. I think the variable name is not proper.

Reply:
there are other scenarios, which cdc output incorrect results. so I suggest to check every value credible. ref: https://www.oceanbase.com/docs/common-oceanbase-database-cn-1000000001052799

Reply:
For issue 1: I replaced != VALUE_ORIGIN::REDO with == VALUE_ORIGIN::PADDING.

For issue 2: I renamed it to has_unreliable_lob_null to make the semantics clearer.

For issue 3: I think that sometimes the old_col generated by CDC for updates can be inaccurate, so I checked old_col again.

If there are still any errors, please let me know, and I will continue to make corrections. Thank you for your review.

Reply:
For issue 3: In fact, not only lob types, but also other types may not output reliable results due to the minimal mode, so you should check each value and then refactor the `has_lob_null`check logic.

Reply:
Thank you for the hint!
In minimal mode, only the columns that have changed will be captured, which means that UPDATE statements may have unreliable NULL values, and the INSERT statements generated by the undo operation of DELETE statements may also have unreliable NULL values. So, I checked each value, changed the variable names and comments, and added as well as passed all unit tests.

Other Information
Fast Reproduce Steps (Required):

Compile and use my modified oblogminer.
Steps to reproduce the behavior:
CREATE TABLE test_table2 (
    id INT PRIMARY KEY NOT NULL,
    value1 INT NOT NULL,
    value2 INT NOT NULL
);
INSERT INTO test_table2 (id, value1, value2) VALUES (1, 10, 20);
UPDATE test_table2 SET value1=30 WHERE id=1;
DELETE FROM test_table2 WHERE id=1;

RESULT
![image](https://github.com/user-attachments/assets/d29e68bd-5368-46a5-a24c-6ad576251a98)
![image](https://github.com/user-attachments/assets/da0445ce-aa0c-4bd1-b587-0b3b35711718)